pip/src/interop.rs

```diff
@@ -55,12 +55,20 @@ impl<T> SourceResolver for ImportResolver<T>
 where
     T: FileSystem,
 {
-    fn resolve<P>(&self, path: P) -> miette::Result<(PathBuf, String)>
+    fn ctx(&mut self) -> &mut SourceResolverContext {
+        &mut self.ctx
+    }
+
+    fn resolve<P>(&mut self, path: P) -> miette::Result<(PathBuf, String), qsc::qasm::io::Error>
```

Comment:
How much do you care about symlinks? Since we don't canonicalize paths in this impl, we won't detect cycles involving symlinks.

FWIW, not "handling" symlinks is the correct behavior to me. There are an infinite number of dumb things you can do with paths so let's not try chasing them. However, we may want to have some kind of depth limit in the paths to prevent the user from shooting themselves in the foot.